js/app.js

```diff
@@ -945,35 +971,24 @@ define([
 				// don't load if it is already loaded or a new concept set
 				if (self.currentConceptSet() && self.currentConceptSet()
 					.id == conceptSetId) {
-					self.currentConceptSetMode(mode);
-					self.currentView(viewToShow);
+          self.currentView(viewToShow);
+          self.currentConceptSetMode(mode);
 					return;
 				}
 				self.currentView('loading');
-				$.ajax({
-					url: config.api.url + 'conceptset/' + conceptSetId,
-					method: 'GET',
-					contentType: 'application/json',
-					error: self.authApi.handleAccessDenied,
-					success: function (conceptset) {
-						$.ajax({
-							url: config.api.url + 'conceptset/' + conceptSetId + '/expression',
-							method: 'GET',
-							contentType: 'application/json',
-							error: self.authApi.handleAccessDenied,
-							success: function (expression) {
-								self.setConceptSet(conceptset, expression.items);
-								self.currentView(viewToShow);
-								var resolvingPromise = self.resolveConceptSetExpression();
-								$.when(resolvingPromise)
-									.done(function () {
-										self.currentConceptSetMode(mode);
-										$('#conceptSetLoadDialog')
-											.modal('hide');
-									});
-							}
-						});
-					}
+				const getConceptSetPromise = conceptSetApi.getConceptSet(conceptSetId);
+				const getExpressionPromise = conceptSetApi.getConceptSetExpression(conceptSetId);
+				$.when(getConceptSetPromise, getExpressionPromise).done((conceptSet, expression) => {
+					self.setConceptSet(conceptSet[0], expression[0].items);
+					self.currentView(viewToShow);
+          var resolvingPromise = self.resolveConceptSetExpression();
+          $.when(resolvingPromise)
+            .done(function () {
+            	self.resolveInclusionConceptCount(conceptSetId);
+              self.currentConceptSetMode(mode);
+              $('#conceptSetLoadDialog')
```

Comment:
Don't use element selectors.  Use the modal databinding, use an observable to set the modal sate: when the call back is done, set the observable to 'false' and that will hide the modal.

Reply:
Btw: I now realize that this is existing code and you aren't actually introducing these element selectors. However, if you can clean this up as part of this PR, that'd be ideal.